### PR TITLE
nrfbsim: Provide a few updated to enable more apps

### PR DESCRIPTION
**Pull request by aescolar**
A few minor updates which enable more apps.
One commit provides a macro defined for ARM some apps need. The other 2 are minor updates to the native simulator and HW models.

* boards nrf_bsim: Add NVIC_GetPendingIRQ() equivalent
* boards nrf_bsim: Provide IRQ_PRIO_LOWEST
* manifest: Update nRF hw models to latest
* native_simulator: Get latest from upstream

### CODE REVIEW

**Review by hermabe — DISMISSED** [review]

**Comment by aescolar** [comment]
CC @ilhanates

**Review by hermabe — APPROVED** [review]

**Review by alexsven — APPROVED** [review]

**Review by rugeGerritsen — APPROVED** [review]

File: boards/native/nrf_bsim/board_irq.h [hunk 1]
```diff
@@ -18,6 +18,7 @@ void nrfbsim_WFE_model(void);
 void nrfbsim_SEV_model(void);
 
 #define IRQ_ZERO_LATENCY	BIT(1) /* Unused in this board*/
+#define IRQ_PRIO_LOWEST		UINT8_MAX
```
Comment (cvinayak) [inline boards/native/nrf_bsim/board_irq.h:21]:
```suggestion
#define IRQ_PRIO_LOWEST  UINT8_MAX
```

**Review by cvinayak — DISMISSED** [review]

File: boards/native/nrf_bsim/board_irq.h (hunk 1 above)
Comment (aescolar) [reply boards/native/nrf_bsim/board_irq.h:21]: you don't like the defines aligned? :)

Comment (cvinayak) [reply boards/native/nrf_bsim/board_irq.h:21]: There is a tab, is it not? ... no. of spaces to align i can not make out in github, but definitely see a tab though.

Comment (cvinayak) [reply boards/native/nrf_bsim/board_irq.h:21]: Tab only allowed at the beginning of lines in Zephyr, atleast that is how compliance used to be.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**